src/utils/PolygonConnector.cpp

```diff
@@ -0,0 +1,271 @@
+    {
+        size_t vert_idx =
+            (dir > 0)?
+            (start.point_idx + 1 + vert_nr) % poly.size()
+            : (static_cast<size_t>(start.point_idx) - vert_nr + poly.size()) % poly.size(); // cast in order to accomodate subtracting
+        if (!first_iter && vert_idx == (end.point_idx + ((dir > 0)? 1 : 0)) % poly.size())
+        {
+            break;
+        }
+        first_iter = false;
+        result.add(poly[vert_idx]);
+    }
+    result.add(end.p());
+}
+
+char PolygonConnector::getPolygonDirection(const ClosestPolygonPoint& from, const ClosestPolygonPoint& to)
+{
+    assert(from.poly == to.poly && "We can only bridge from one polygon from the other if both connections depart from the one polygon!");
+    ConstPolygonRef poly = *from.poly;
+    if (from.point_idx == to.point_idx)
+    {
+        const Point prev_vert = poly[from.point_idx];
+        const coord_t from_dist2 = vSize2(from.p() - prev_vert);
+        const coord_t to_dist2 = vSize2(to.p() - prev_vert);
+        return (to_dist2 > from_dist2)? 1 : -1;
+    }
+    // TODO: replace naive implementation by robust implementation
+    // naive idea: there are less vertices in between the connection points than around
+    const size_t a_to_b_vertex_count = (static_cast<size_t>(to.point_idx) - from.point_idx + poly.size()) % poly.size(); // cast in order to accomodate subtracting
+    if (a_to_b_vertex_count > poly.size() / 2)
+    {
+        return -1; // from a to b is in the reverse direction as the vertices are saved in the polygon
+    }
+    else
+    {
+        return 1; // from a to b is in the same direction as the vertices are saved in the polygon
+    }
+}
+
+std::optional<PolygonConnector::PolygonBridge> PolygonConnector::getBridge(ConstPolygonRef from_poly, std::vector<Polygon>& to_polygons)
+{
+    std::optional<PolygonConnector::PolygonConnection> connection = getConnection(from_poly, to_polygons);
+    if (!connection || connection->getDistance2() > max_dist * max_dist)
+    {
+        return std::optional<PolygonConnector::PolygonBridge>();
+    }
+
+    // try to get the other connection forward
+    std::optional<PolygonConnector::PolygonConnection> second_connection = PolygonConnector::getSecondConnection(*connection, line_width);
+    if (!second_connection)
+    {
+        // try to get a bridge with the connections on both sides of the initially calculated connection
+        second_connection = PolygonConnector::getSecondConnection(*connection, line_width / 2);
+        if (second_connection)
+        {
+            connection = PolygonConnector::getSecondConnection(*connection, line_width);
+        }
+    }
+    if (connection && second_connection)
+    {
+        PolygonBridge result;
+        result.a = *connection;
+        result.b = *second_connection;
+        // ensure that b is always the right connection and a the left
+        Point a_vec = result.a.to.p() - result.a.from.p();
```

Comment:
If this constraint  "ensure that b is always the right.."  then we probably need to add to the description

Reply:
It is already in the documentation of PolygonBridge: `--> connection a is always the left one`